src/psyclone/psyir/symbols/symbol.py

```diff
@@ -244,6 +244,98 @@ def __init__(self, name, visibility=DEFAULT_VISIBILITY, interface=None):
             # Use the setter as it checks the variables validity
             self.interface = interface
 
+    def copy(self, interface=None, visibility=None):
+        '''Create and return a copy of this object. Any references to the
+        original will not be affected so the copy will not be referred
+        to by any other object.
+
+        :param interface: optional value with which to override this object's \
+                interface in the new object.
+        :type interface: :py:class:`psyclone.psyir.symbols.SymbolInterface`
+        :param visibility: optional value with which to override this object's\
+                visibility in the new object.
+        :type visibility: :py:class:`psyclone.psyir.symbols.Symbol.Visibility`
+
+        :returns: A symbol object with the same properties as this \
+                  symbol object.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        return self._base_copy([], {}, interface=interface,
+                               visibility=visibility)
+
+    def _base_copy(self, args, kwargs, interface=None, visibility=None):
```

Comment:
Again, I need a bit of context about why this is needed instead of the original implementation.